openmc/polynomial.py

```diff
@@ -75,7 +74,68 @@ def order(self):
     def __call__(self, r):
         import openmc.lib as lib
         if isinstance(r, Iterable):
-            return [np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r_i))
+            return [np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r_i / self.radius))
                     for r_i in r]
         else:
-            return np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r))
+            return np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r / self.radius))
+
+
+class Zernike(Polynomial):
+    r"""Create Zernike polynomials given coefficients and domain.
+    
+    The azimuthal Zernike polynomials are defined as in :class:`ZernikeFilter`.
+    
+    Parameters
+    ----------
+    coef : Iterable of float
+        A list of coefficients of each term in radial only Zernike polynomials
+    radius : float
+        Domain of Zernike polynomials to be applied on. Default is 1.
+        
+    Attributes
+    ----------
+    order : int
+        The maximum (even) order of Zernike polynomials.
+    radius : float
+        Domain of Zernike polynomials to be applied on. Default is 1.
+    theta : float 
+        Azimuthal of Zernike polynomial to be applied on. Default is 0.    
+    norm_coef : iterable of float
+        The list of coefficients of each term in the polynomials after
+        normailization.
+    """
+    def __init__(self, coef, radius=1):
+        super().__init__(coef)
+        r"""Solve order from number of coefficients
+        N = (order + 1)(order + 2) / 2
+        """
```

Comment:
This should just be a normal comment
```suggestion
        # Solve order from number of coefficients
        # N = (order + 1)(order + 2) / 2
```